assets/js/googlesitekit/datastore/site/cache.js

```diff
@@ -0,0 +1,90 @@
+/**
+ * `core/site` data store: client side cache.
+ *
+ * Site Kit by Google, Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import invariant from 'invariant';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { setItem } from '../../../googlesitekit/api/cache';
+
+// Actions.
+const CACHE_SET_ITEM = 'CACHE_SET_ITEM';
+
+const baseInitialState = {};
+
+const baseActions = {
+	/**
+	 * Yields to setItem function which sets cached data using a key.
+	 *
+	 * @since n.e.x.t
+	 * @see setItem
+	 *
+	 * @param {string}  key              Name of cache key.
+	 * @param {*}       value            Value to store in the cache.
+	 * @param {Object}  args             Optional object containing ttl, timestamp and isError keys.
+	 * @param {number}  [args.ttl]       Optional. Validity of the cached item in seconds.
+	 * @param {number}  [args.timestamp] Optional. Timestamp when the cached item was created.
+	 * @param {boolean} [args.isError]   Optional. Whether the cached item is an error.
```

Comment:
For an arguments object with properties that are all optional, we use `options` as the name for this kind of object as a convention. `args` could contain optional properties but implies that at least one or more are required.